### PR TITLE
[FIX] 0025216: PHP7.3: "continue" targeting switch is equivalent to "…

### PR DESCRIPTION
**Pull request by chfsx**
…break". Did you mean to use "continue 2"? and one count() fix

See https://mantis.ilias.de/view.php?id=25216

### CODE REVIEW

**Comment by corro** [comment]
As far as I understand the issue, this change might alter the behavior of the affected code: `continue` behaves like `break` in PHP, therefore any remaining code in the surrounding `for` loop is executed after the `switch` statement. With this change here this is no longer the case.

It may well be that this was the authors intention in the first place for using `continue` instead of `break`. I just wanted to point out that this change might potentially break things.

**Comment by chfsx** [comment]
@corro yes I did see that and tried to assume the authors intention in this PR. But the maintainers must check if the behaviour is still the same and intentioned with these changes.

**Comment by alex40724** [comment]
Thanks for the PR. I hand over the ilPDSelectedItemsBlockGUI change to @mjansenDatabay since this may be relevant or not for the forum notification handling in his code. The rest belongs to @smeyer-ilias as far as I can see.

**Comment by alex40724** [comment]
I set @mjansenDatabay as reviewer and kept the assignment to @smeyer-ilias as it seems to be feasible that he makes the final merge.

File: Services/PersonalDesktop/ItemsBlock/classes/class.ilPDSelectedItemsBlockGUI.php
Comment (smeyer-ilias) [inline Services/PersonalDesktop/ItemsBlock/classes/class.ilPDSelectedItemsBlockGUI.php]: @mjansenDatabay : Is it necessary to execute the following forum lines or is a "continue 2" correct here.

**Review by smeyer-ilias — COMMENTED** [review]
Thanks Fabian! I will merge after @mjansenDatabay mjansen has confirmed.

File: Services/PersonalDesktop/ItemsBlock/classes/class.ilPDSelectedItemsBlockGUI.php
Comment (mjansenDatabay) [reply Services/PersonalDesktop/ItemsBlock/classes/class.ilPDSelectedItemsBlockGUI.php]: The code MUST only be executed for courses and groups, NOT for the default. So a `continue 2;` should be okay here.

**Review by mjansenDatabay — APPROVED** [review]

**Comment by smeyer-ilias** [comment]
Thanks for fixing this!

**Comment by corro** [comment]
Thank you for the fixes! Since ILIAS 5.4 supports PHP 7.3 too (and suffers from the same errors), shouldn't these be backported? As far as I can see, this is not yet the case.

**Comment by chfsx** [comment]
@smeyer-ilias It is currently a problem in 5.4: https://docu.ilias.de/goto_docu_frm_1875_5784.html

**Comment by smeyer-ilias** [comment]
I could cherry-pick this to 5.4
But install.md in 5.4 says:

## Supported Platforms

Please note that different configurations SHOULD be possible, but it might be harder to find someone who can help when things go south. You SHALL NOT use a different configuration unless you are an experienced system administrator.


### Server

  * Server OS: Linux
  * Web Server: Apache 2.4 (mod_php, php-fpm)
  * Databases: MySQL/MariaDB 5.6 and 5.7 and Galera (experimental), PostgreSQL 9.x
  *** PHP: Version 7.0, 7.1 and 7.2 are supported**



| ILIAS Version   | PHP Version                           |
|-----------------|---------------------------------------|
| 5.4.x           | 7.0.x, 7.1.x, 7.2.x, 7.3.x            |

**Comment by smeyer-ilias** [comment]
Is PHP 7.3 really supported?

**Comment by corro** [comment]
Yes, ILIAS 5.4 officially supports PHP 7.3. Here the corresponding feature wiki entry:
https://docu.ilias.de/goto_docu_wiki_wpage_5047_1357.html

**Comment by corro** [comment]
@smeyer-ilias We regularly struggle with this error, since our Docker images use ILIAS 5.4 with PHP 7.3 by default. Could you please consider backporting this fix to 5.4?

**Comment by smeyer-ilias** [comment]
Sorry, I had overseen your answer. 
Merged to 5.4.

**Pull request merged**
